### PR TITLE
TST: split up pandas/tests/test_multilevel.py

### LINKED ISSUE #23782: TST: split up pandas/tests/test_multilevel.py
we should move indexing routines from ``pandas/tests/test_multilevel.py`` to 

``pandas/tests/indexing/test_multiindex.py``

things that  have test names that match ``getitem`` or ``xs``.

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ ] closes #23782
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [n/a ] whatsnew entry

@jreback 
1. i've used fixtures for `frame` and `ymd` instead of copying across the test base class.
2. not moved tests with names matching `getitem` from the `TestSorted` class.
3. frame used in `test_getitem_int` renamed to _frame to avoid clash with fixture.

### CODE REVIEW

**Comment by jreback** [comment]
ok, can you merge master. i think something is failing as well (the 2.7 should not be a problem any longer). ideally also see if you can move other indexing tests (that you commented about above), even if you have to slightly duplicate whats in the the base class (but do it as fixtures).

**Comment by simonjayhawkins** [comment]
> i think something is failing as well (the 2.7 should not be a problem any longer)

could have been zip or range...  now added to `from pandas.compat ...`

**Comment by jreback** [comment]
you need to isort imports

File: pandas/tests/indexing/test_multiindex.py [hunk 1]
```diff
@@ -1,19 +1,47 @@
...
-    period_range)
+    DataFrame, Index, MultiIndex, Panel, Period, Series, Timestamp, concat,
+    date_range, period_range, read_csv)
+import pandas.core.common as com
 from pandas.tests.indexing.common import _mklbl
 from pandas.util import testing as tm
 
 
+@pytest.fixture
+def frame():
```
Comment (jreback) [inline pandas/tests/indexing/test_multiindex.py]: i had left some comments on the fixture names

Comment (simonjayhawkins) [reply pandas/tests/indexing/test_multiindex.py]: sorry @jreback , either i'm not looking at the right comment or i've misunderstood something. can you clarify?

Comment (jreback) [reply pandas/tests/indexing/test_multiindex.py]: yes, can you give these meaningful names

File: pandas/tests/test_multilevel.py [hunk 2]
```diff
@@ -239,57 +237,6 @@ def test_repr_name_coincide(self):
...
-        # don't segfault, GH #495
-        # out of bounds access
-        pytest.raises(IndexError, s.__getitem__, len(self.ymd))
-
-        # generator
-        result = s[(x > 0 for x in s)]
-        expected = s[s > 0]
-        tm.assert_series_equal(result, expected)
-
     def test_series_setitem(self):
```
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/tests/test_multilevel.py]: i've only moved things that have test names that match `getitem` or `xs`.

[the remark this answers is not in the document]
Comment (jreback) [reply pandas/tests/test_multilevel.py]: you can move setitem as well

[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/tests/test_multilevel.py]: - [ ]  test_get_loc_single_level
- [ ]  test_fancy_slice_partial
- [ ] test_partial_set
- [ ] test_partial_ix_missing
- [ ] test_int_series_slicing
- [ ] test_mixed_depth_get
- [ ] test_mixed_depth_insert
- [ ] test_dataframe_insert_column_all_na
- [ ] test_set_column_scalar_with_ix
- [ ] test_indexing_ambiguity_bug_1678
- [ ] test_nonunique_assignment_1750
- [ ] test_indexing_over_hashtable_size_cutoff
- [ ] test_iloc_mi

as well as the tests with names that match setitem?

**Comment by jreback** [comment]
yes anything that looks like an indexing tesr

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Comment by jreback** [comment]
it might make sense to now split pandas/tests/indexing/test_multiindex -> pandas/test/indexing/multiindex/

with a series of files under multiindex logically split.

**Comment by simonjayhawkins** [comment]
@jreback i recall seeing some discussion on moving towards the functional idiom of pytest but can't seem to find it.

i'll raise a follow on PR and on splitting the test i'll drop the classes as has been done in https://github.com/pandas-dev/pandas/blob/master/pandas/tests/arrays/test_period.py

**Pull request merged**